presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/ElasticsearchQueryRunner.java

```diff
@@ -0,0 +1,165 @@
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.facebook.presto.tests.TestingPrestoClient;
+import com.facebook.presto.tpch.TpchPlugin;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Resources;
+import io.airlift.json.JsonCodec;
+import io.airlift.log.Logger;
+import io.airlift.log.Logging;
+import io.airlift.tpch.TpchTable;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.facebook.presto.elasticsearch.ElasticsearchTableDescriptionSupplier.createElasticsearchTableDescriptions;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.airlift.units.Duration.nanosSince;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.Assert.assertNotNull;
+
+public final class ElasticsearchQueryRunner
+{
+    private ElasticsearchQueryRunner()
+    {
+    }
+
+    private static final Logger log = Logger.get("TestQueries");
+    private static final String TPCH_SCHEMA = "tpch";
+
+    public static DistributedQueryRunner createElasticsearchQueryRunner(EmbeddedElasticsearch embeddedElasticsearch, TpchTable<?>... tables)
+            throws Exception
+    {
+        return createElasticsearchQueryRunner(embeddedElasticsearch, ImmutableList.copyOf(tables));
+    }
+
+    public static DistributedQueryRunner createElasticsearchQueryRunner(EmbeddedElasticsearch embeddedElasticsearch, Iterable<TpchTable<?>> tables)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = new DistributedQueryRunner(createSession(), 2);
+
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            embeddedElasticsearch.start();
+
+            Map<SchemaTableName, ElasticsearchTableDescription> tableDescriptions = createTableDescriptions(queryRunner.getCoordinator().getMetadata(), tables);
+
+            installElasticsearchPlugin(embeddedElasticsearch, queryRunner, tables, tableDescriptions);
+
+            TestingPrestoClient prestoClient = queryRunner.getClient();
+
+            log.info("Loading data...");
+            long startTime = System.nanoTime();
+            for (TpchTable<?> table : tables) {
+                loadTpchTopic(embeddedElasticsearch, prestoClient, table);
+            }
+            log.info("Loading complete in %s", nanosSince(startTime).toString(SECONDS));
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
```

Comment:
Is `Throwable` too broad? Please use `Exception`

Reply:
get it, will fix